pkg/controller/cloud/node_controller.go

```diff
@@ -313,11 +313,16 @@ func (cnc *CloudNodeController) MonitorNode() {
 }
 
 func (cnc *CloudNodeController) UpdateCloudNode(_, newObj interface{}) {
-	if _, ok := newObj.(*v1.Node); !ok {
+	node, ok := newObj.(*v1.Node)
+	if !ok {
 		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
 		return
 	}
-	cnc.AddCloudNode(newObj)
+
+	cloudTaint := getCloudTaint(node.Spec.Taints)
```

Comment:
FYI there's a similar check here https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L327-L331.

Reply:
oh I see now, should read the PR description before commenting ;)

Reply:
:)

Reply:
I don't love that we're checking taints twice here, though it is unlikely to happen since most nodes should register from the OnAdd event. I would suggest just increasing the log level [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/cloud/node_controller.go#L329) but it seems like that log message could be useful when OnAdd events actually fail 🤔